modin/numpy/arr.py

```diff
@@ -1469,6 +1559,283 @@ def sum(
                 + initial
             )
 
+    def all(self, axis=None, out=None, keepdims=None, *, where=True):
```

Comment:
Is `where` a mask on the output (like with ge, or is it supposed to be a mask on the input like with mean)?

Reply:
It seems to be a mask on the input:
```
>>> numpy.array([[0,1],[1,0]]).all(where=[[0,1],[1,0]])
True
```

I'm not sure how to efficiently implement `where` masking properly in the `axis=None` case if we're doing a double reduction with `self.any(axis=0).any(axis=1)`. Any suggestions?

Reply:
Sounds good - we can just do something like `where.where(self, True)` for `all` and `where.where(self, False)` for `any`.